sstables/sstables.cc

```diff
@@ -1778,6 +1778,62 @@ sstable_writer sstable::get_writer(const schema& s, uint64_t estimated_partition
     return sstable_writer(*this, s, estimated_partitions, cfg, enc_stats, pc, shard);
 }
 
+future<uint64_t> sstable::validate(schema_ptr schema, reader_permit permit, const io_priority_class& pc, abort_source& abort,
```

Comment:
Do we need the schema here? Since sstable was constructed with a schema.

This could be generic code (validating a fragment stream), but I also see it wanting to cross-reference the index, so maybe not.

Reply:
Please document the int64_t returned (or return a struct with a named member).

Reply:
Ah, it's documented.